File: src/main/java/ma/emsi/PFAcabinetsDeMedcin/Medecin/Entities/Cabinet.java
package ma.emsi.PFAcabinetsDeMedcin.Medecin.Entities;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OneToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor@NoArgsConstructor
@Entity
public class Cabinet {
    @Id
    private Long id;
    @OneToOne
    private Medcin med;
    @OneToMany
    private List<Disponibilite> disponibilites;
    private String localisation;
}

File: medcinPatient-network-ui/src/app/services/models/cabinet.ts
/* tslint:disable */
/* eslint-disable */
import { Disponibilite } from '../models/disponibilite';
import { Medcin } from '../models/medcin';
export interface Cabinet {
  disponibilites?: Array<Disponibilite>;
  id?: number;
  localisation?: string;
  med?: Medcin;
}
